mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1242,13 +1297,52 @@ impl TakerSwap {
             tx_hash,
         };
 
-        Ok((Some(TakerSwapCommand::WaitForTakerPaymentSpend), vec![
-            TakerSwapEvent::TakerPaymentSent(tx_ident),
-        ]))
+        let mut swap_events = vec![TakerSwapEvent::TakerPaymentSent(tx_ident)];
+        if self.ctx.use_watchers() {
+            let preimage_fut = self.taker_coin.create_taker_spends_maker_payment_preimage(
+                &self.r().maker_payment.as_ref().unwrap().tx_hex,
+                self.maker_payment_lock.load(Ordering::Relaxed) as u32,
+                self.r().other_maker_coin_htlc_pub.as_slice(),
+                &self.r().secret_hash.0,
+                &self.unique_swap_data()[..],
+            );
+
+            // If the watcher message can not be sent, the swap still continues
+            if let Ok(preimage) = preimage_fut.compat().await {
+                let preimage_hex = Some(preimage.tx_hex());
```

Comment:
Why this preimage_hex is not so necessary for watching swap events?

Reply:
preimage_hex is actually required for the watcher. My intention was that if an error occurs and the taker can't compute the preimage for some reason, it should just not broadcast the watcher message, but continue working on its own taker swap. I didn't want an error due to the watcher message to block the actual swap itself, even though use_watchers configuration is set to true. What's your opinion on this @artemii235?

Reply:
@caglaryucekaya how about just logging the information about the absence of preimage to the user?
What I mean is, there is no preimage, it could be informative for people.